code/game/objects/effects/spawners/random/trash.dm

```diff
@@ -0,0 +1,237 @@
+		/obj/item/reagent_containers/glass/rag = 2,
+		/obj/item/reagent_containers/glass/bucket = 2,
+		/obj/effect/decal/cleanable/blood/old = 2,
+		/obj/structure/mopbucket = 2,
+	)
+
+/obj/effect/spawner/random/trash/graffiti
+	name = "random graffiti spawner"
+	icon_state = "rune"
+	loot = list(/obj/effect/decal/cleanable/crayon)
+	var/graffiti_icons = list(
+		"rune1", "rune2", "rune3", "rune4", "rune5", "rune6",
+		"amyjon", "face", "matt", "revolution", "engie", "guy",
+		"end", "dwarf", "uboa", "body", "cyka", "star", "poseur tag",
+		"prolizard", "antilizard", "danger", "firedanger", "electricdanger",
+		"biohazard", "radiation", "safe", "evac", "space", "med", "trade", "shop",
+		"food", "peace", "like", "skull", "nay", "heart", "credit",
+		"smallbrush", "brush", "largebrush", "splatter", "snake", "stickman",
+		"carp", "ghost", "clown", "taser", "disk", "fireaxe", "toolbox",
+		"corgi", "cat", "toilet", "blueprint", "beepsky", "scroll", "bottle",
+		"shotgun", "arrow", "line", "thinline", "shortline", "body", "chevron",
+		"footprint", "clawprint", "pawprint",
+	)
+	color = COLOR_WHITE //sets the color of the graffiti (used for mapedits)
+	var/random_color = TRUE //whether the graffiti will spawn with a random color (used for mapedits)
+	var/random_icon = TRUE // whether the graffiti will spawn with the same icon
+
+/obj/effect/spawner/random/trash/graffiti/proc/select_graffiti(graffiti_decal)
+	var/obj/effect/decal/cleanable/crayon/decal = graffiti_decal
+	color = random_color && "#[random_short_color()]" || color
+	icon_state = random_icon && pick(graffiti_icons) || icon_state
+
+	decal.add_atom_colour(color, FIXED_COLOUR_PRIORITY)
+	decal.icon_state = icon_state
+
+/obj/effect/spawner/random/trash/mopbucket
+	name = "mop bucket spawner"
+	icon_state = "mopbucket"
+	spawn_loot_count = 2
+	spawn_loot_double = FALSE
+	loot = list(
+		/obj/structure/mopbucket = 10,
+		/obj/item/mop = 5,
+		/obj/item/clothing/suit/caution = 3,
+		/obj/item/reagent_containers/glass/bucket = 1,
+		/obj/item/reagent_containers/glass/bucket/wooden = 1,
+	)
+
+/obj/effect/spawner/random/trash/caution_sign
+	name = "caution sign spawner"
+	icon_state = "caution"
+	loot = list(
+		/obj/item/clothing/suit/caution = 40,
+		/obj/structure/holosign/wetsign = 5,
+		/obj/structure/holosign/barrier = 3,
+		/obj/structure/holosign/barrier/wetsign = 2,
+	)
+
+
+/obj/effect/spawner/random/structure/bucket
+	name = "bucket spawner"
+	icon_state = "caution"
+	loot = list(
+		/obj/item/reagent_containers/glass/bucket,
+		/obj/item/reagent_containers/glass/bucket/wooden,
+	)
+
+/obj/effect/spawner/random/trash/soap
+	name = "soap spawner"
+	icon_state = "soap"
+	loot = list(
+		/obj/item/soap = 25,
+		/obj/item/bikehorn/rubberducky = 20,
+		/obj/item/soap/homemade = 20,
+		/obj/item/soap/deluxe = 15,
+		/obj/item/soap/nanotrasen = 10,
+		/obj/item/food/urinalcake = 5,
+		/obj/item/bikehorn/rubberducky/plasticducky = 4,
+		/obj/item/soap/omega = 1,
```

Comment:
not sure about this randomly spawning, the crazy recipe to crafting it is meant to be half the fun of it

Reply:
Removed with https://github.com/tgstation/tgstation/pull/60522/commits/f3dadfc849c3f7b4e5589838cd4364a6e6fe4e25